### PR TITLE
Removes wallshove knockdowns

### PR DESCRIPTION
**Pull request by Moribox**
## About The Pull Request
This simply makes it when a player shoves another player / humanoid into a wall, they no longer get knocked down.

## Why It's Good For The Game
The combat isn't really made with this sort of stun in mind, and wallshoves have on several occasions resulted in scenarios where a fight that should otherwise be in one sides favor is instantly lost to another side due to " oh well i'm stunned guess I die now."

## Changelog
:cl:
del: Removed the ability to knock down players / humanoids with wallshove.
/:cl:

### CODE REVIEW

**Comment by LanceSmites328** [comment]
Add a line after line 315 in carbon_defense.dm that checks for city map and returns if true.
Something like
[315] (if statement)
[316]     if(SSmaptype.maptype == "city")
[317]         return
[318]     target.Knockdown(SHOVE_KNOCKDOWN_SOLID)
[319]     ...
As opposed to simply setting the knockdown to 0.
You should also go to combat.dm and add a similar if-statement before line 97, once again checking for citymap. If this if-statement does not work, you should likely remove lines 90+ from this file.
It should look something like
[96] TEST_ASSERT_EQUAL(victim.loc.x, run_loc_floor_bottom_left.x + 2, "Victim was moved after being pushed against a wall")
[97] if(SSmaptype.maptype == "city")
[98]     TEST_ASSERT(!victim.has_status_effect(STATUS_EFFECT_KNOCKDOWN), "Victim was knocked down after being pushed against a wall on city map")
[99]     TEST_ASSERT_EQUAL(victim.get_active_held_item(), toolbox, "Victim dropped toolbox after being pushed against a wall on city map")
[100] else
[101]     TEST_ASSERT(victim.has_status_effect(STATUS_EFFECT_KNOCKDOWN), "Victim was not knocked down after being pushed against a wall")
[102]    TEST_ASSERT_EQUAL(victim.get_active_held_item(), null, "Victim didn't drop toolbox after being pushed against a wall")

**Comment by LanceSmites328** [comment]
Mind you, this should all be done in place of the duration change, that should be reverted.

**Comment by Moribox** [comment]
Should be functional now! literally all credit for this goes to redacted / lance. turns out setting it to 0 does have consiquences!

File: code/modules/unit_tests/combat.dm [hunk 1]
```diff
@@ -92,7 +92,8 @@
 
 	// Second disarm, victim was against wall and should be down
 	victim.attack_hand(attacker)
-
+		if(SSmaptype.maptype == "city")
```
Comment (LanceSmites328) [inline code/modules/unit_tests/combat.dm:95]:
```suggestion
	if(SSmaptype.maptype == "city")
```

**Review by LanceSmites328 — CHANGES_REQUESTED** [review]
Rebase/squash was performed improperly and has gained a commit, or commits, included in Drifting Fox. Please revert said commits.

**Comment by Moribox** [comment]
Ah shoot, I will amend this, must have accidentally selected the same branch I did wall on when I started working on fox.

**Comment by Moribox** [comment]
Hopefully fixed!

**Review by LanceSmites328 — APPROVED** [review]
I went through so much to test the unit test and make sure it functioned properly. I believe it does.

**Pull request merged**